src/receive_imf.rs

```diff
@@ -1804,6 +1805,19 @@ async fn lookup_chat_by_reply(
     };
     let parent_chat = Chat::load_from_db(context, parent_chat_id).await?;
 
```

Comment:
Off-topic, not related to this PR: Thinking about the `lookup_chat_by_reply()` logic, I think that we shouldn't assign by reply for protected chats. I.e. add:

```rust
    // Don't allow assigning messages from another sender to a protected chat,
    // because we only want trusted messages in there.
    // The exception is undecryptable messages because we can't see the Chat-Group-Id
    // (since it's encrypted), so we do allow to put those into a protected chat.
    if parent_chat.is_protected() || mime_parser.decrypting_failed {
        return Ok(None);
    }
```

I can do this in a follow-up PR, or you are welcome do do it.

Reply:
`lookup_chat_by_reply()` isn't about another sender, it's for the case of an absent `Chat-Group-Id`. Another sender, if they know the chat group id somehow, can easily send messages to the chat, so rather we should check if they are actually a member. So the check for `is_protected()` here doesn't help IMO, but it prevents a member from answering to the chat from other MUA using the same (verified) key which should be fine. E.g. it may be another device of the chat member where they can't use Delta Chat for some reason.